core/src/main/kotlin/knote/PageManagerImpl.kt

```diff
@@ -88,16 +106,34 @@ internal class PageManagerImpl(
         page.compiledScript = pageScript
         if (pageScript == null) {
             logger.error("evaluation failed for file $file")
-            reports.forEach {
-                logger.error { it }
+            reports.forEach { report ->
+                val path = report.sourcePath?.let { "$it: " } ?: ""
+                val location = report.location?.posToString()?.let { "$it: " } ?: ""
+                val messageString = "$path$location${report.message}"
+                when (report.severity) {
+                    ScriptDiagnostic.Severity.FATAL -> logger.error { "FATAL: $messageString" }
+                    ScriptDiagnostic.Severity.ERROR -> logger.error { messageString }
+                    ScriptDiagnostic.Severity.WARNING -> logger.warn { messageString }
+                    ScriptDiagnostic.Severity.INFO -> logger.info { messageString }
+                    ScriptDiagnostic.Severity.DEBUG -> logger.debug { messageString }
```

Comment:
wow, error messaging too nice

Reply:
the GUI can have a similar switch for colors